org.osgi.test.junit5.cm/src/test/java/org/osgi/test/junit5/cm/ConfigAnnotationNested.java

```diff
@@ -0,0 +1,37 @@
+package org.osgi.test.junit5.cm;
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.osgi.service.cm.Configuration;
+import org.osgi.test.common.annotation.config.CmAction;
+import org.osgi.test.common.annotation.config.InjectConfiguration;
+
+@ExtendWith(ConfigurationExtension.class)
+@InjectConfiguration(pid = "sub.0")
```

Comment:
Where is this configuration being injected?

Reply:
"into the framework;)". So you can separate test configurations using nested tests with inner classes

Reply:
Changes into @WithConfiguration